misk/src/test/kotlin/misk/web/ContentBasedDispatchTest.kt

```diff
@@ -30,13 +37,21 @@ internal class ContentBasedDispatchTest {
 
     data class Packet(val message: String)
 
+    data class ComplexPacket(val message: String, val composedBy: String = "default") {
```

Comment:
What this test is covering is a bit hard to understand. I'd create separate tests for "use generic unmarshalling of types with valueOf methods" and "picks the appropriate unmarshaller based on incoming content type". The first could just be tested at the unmarshaller level (i.e. be in a simpler more isolated test class), the second is more appropriate to content based dispatch.

Reply:
Yeah, I suppose the test outgrew the parameters of this task set. I'll split it out.